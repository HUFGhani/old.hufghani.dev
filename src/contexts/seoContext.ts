import { createContext, useContext } from 'react'

interface SEOContextType {
  data: {
    site: {
      siteMetadata: {
        author: string
        url: string
        twitter: string
        titleAlt: string
        siteUrl: string
        siteLanguage: string
        shortName: string
        pathPrefix: string
        description: string
      }
    }
  }
}

export const SEOContext = createContext<SEOContextType | undefined>(undefined)

export const useSEOContext = () => useContext<SEOContextType | undefined>(SEOContext)
