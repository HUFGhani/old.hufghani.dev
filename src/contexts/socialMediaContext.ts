import { createContext, useContext } from 'react'
import SocialMediaInterface from '../interface/socialMediaInterface'

export const SocialMediaContext = createContext<SocialMediaInterface | undefined>(undefined)

export const useSocialMediaContext = () => useContext<SocialMediaInterface | undefined>(SocialMediaContext)
