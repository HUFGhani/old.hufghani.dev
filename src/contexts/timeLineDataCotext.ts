import { createContext, useContext } from 'react'
import TimelineItemInterface from '../interface/timelineItemInterface'

export const TimeLineDataCotext = createContext<TimelineItemInterface | undefined>(undefined)

export const useTimeLineDataCotext = () => useContext<TimelineItemInterface | undefined>(TimeLineDataCotext)
