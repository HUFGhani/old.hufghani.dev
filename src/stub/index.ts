export const SiteMetaData = {
  author: `Hamza Umar Farooq Ghani`,
  url: `https://hufghani.dev`,
  twitter: `@the_ghani`,
  titleAlt: `Hamza Ghani - Personal Website`,
  siteUrl: `https://hufghani.dev`,
  siteLanguage: `en`,
  shortName: `HUFGHANI`,
  pathPrefix: `/`,
  description: `personal Website`,
}
