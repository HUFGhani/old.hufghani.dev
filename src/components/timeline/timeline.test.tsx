import { ChakraProvider, CSSReset } from '@chakra-ui/react'
import { fireEvent, render } from '@testing-library/react'
import React from 'react'
import Timeline from '.'
import { SEOContext } from '../../contexts/seoContext'
import { SocialMediaContext } from '../../contexts/socialMediaContext'
import { TimeLineDataCotext } from '../../contexts/timeLineDataCotext'
import { SiteMetaData, SocialMediaData, TimelineData } from '../../stub'
import Header from '../header'

function renderTimeLine() {
  return render(
    <ChakraProvider>
      <CSSReset />
      <SEOContext.Provider value={SiteMetaData}>
        <SocialMediaContext.Provider value={SocialMediaData}>
          <TimeLineDataCotext.Provider value={TimelineData}>
            <Header />
            <Timeline />
          </TimeLineDataCotext.Provider>
        </SocialMediaContext.Provider>
      </SEOContext.Provider>
    </ChakraProvider>
  )
}

describe(`<Timeline/>`, () => {
  it(`should render Footer`, () => {
    const { asFragment } = renderTimeLine()
    expect(asFragment()).toMatchSnapshot()
  })
  it(`should render correctly and switch the theme from light to dark`, () => {
    const { getByLabelText } = renderTimeLine()
    expect(document.querySelector(`body`)).toHaveClass(`chakra-ui-light`)
    fireEvent.click(getByLabelText(`Hamburger Menu`))
    fireEvent.click(getByLabelText(`Toggle Theme`))
    expect(document.querySelector(`body`)).toHaveClass(`chakra-ui-dark`)
  })
})
