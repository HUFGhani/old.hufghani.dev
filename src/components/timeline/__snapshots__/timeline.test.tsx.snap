// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline/> should render Footer 1`] = `
<DocumentFragment>
  <nav
    class="css-ygbaey"
  >
    <div
      class="css-70qvj9"
    />
    <button
      aria-label="Hamburger Menu"
      class="chakra-button css-1ccmbpd"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-n5gr1q"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
          fill="currentColor"
        />
      </svg>
    </button>
  </nav>
  <div
    class="timeline-container css-fv0w0a"
  >
    <div
      class="timeline-item css-0"
    >
      <div
        class="timeline-item-content css-wlpogo"
      >
        <span
          class="tag css-0"
          style="background: rgb(1, 143, 105);"
        >
          medium
        </span>
        <time
          class="css-0"
        >
          March 03 2017
        </time>
        <p
          class="chakra-text css-0"
        >
          Wrote my first blog post ever on Medium
        </p>
        <span
          class="circle css-1asj0mp"
        />
      </div>
    </div>
    <div
      class="timeline-item css-0"
    >
      <div
        class="timeline-item-content css-wlpogo"
      >
        <span
          class="tag css-0"
          style="background: rgb(1, 143, 105);"
        >
          medium
        </span>
        <time
          class="css-0"
        >
          March 03 2017
        </time>
        <p
          class="chakra-text css-0"
        >
          Wrote my first blog post ever on Medium
        </p>
        <span
          class="circle css-1asj0mp"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
